test/SILGen/moveonly_escaping_closure.swift

```diff
@@ -0,0 +1,1583 @@
+// CHECK:   [[CHECK:%.*]] = mark_must_check [assignable_but_not_consumable] [[ACCESS]]
+// CHECK:   [[LOADED:%.*]] = load [take] [[CHECK]]
+// CHECK:   apply {{%.*}}([[LOADED]])
+// CHECK:   end_access [[ACCESS]]
+//
+// CHECK:   [[READ_ACCESS:%.*]] = begin_access [read] [unknown] [[PROJECT]]
+// CHECK:   [[CHECK:%.*]] = mark_must_check [no_consume_or_assign] [[READ_ACCESS]]
+// CHECK:   [[LOADED_READ:%.*]] = load [copy] [[CHECK]]
+// CHECK:   [[DEINIT_ACCESS:%.*]] = begin_access [deinit] [unknown] [[PROJECT]]
+// CHECK:   [[CHECK:%.*]] = mark_must_check [assignable_but_not_consumable] [[DEINIT_ACCESS]]
+// CHECK:   [[LOADED_TAKE:%.*]] = load [take] [[CHECK]]
+// CHECK:   apply {{%.*}}([[LOADED_READ]], [[LOADED_TAKE]])
+// CHECK:   end_access [[DEINIT_ACCESS]]
+// CHECK:   destroy_value [[LOADED_READ]]
+// CHECK:   end_access [[READ_ACCESS]]
+// CHECK: } // end sil function '$s16moveonly_closure026testLocalVarClosureCaptureE0yyFyycfU_'
+func testLocalVarClosureCaptureVar() {
+    var x = SingleElt()
+    x = SingleElt()
+    var f = {
+        borrowVal(x)
+        consumeVal(x) // expected-error {{'x' was consumed but it is illegal to consume a noncopyable mutable capture of an escaping closure. One can only read from it or assign over it}}
+        consumeVal(x) // expected-error {{'x' was consumed but it is illegal to consume a noncopyable mutable capture of an escaping closure. One can only read from it or assign over it}}
+        borrowConsumeVal(x, x) // expected-error {{'x' was consumed but it is illegal to consume a noncopyable mutable capture of an escaping closure. One can only read from it or assign over it}}
+        // expected-error @-1 {{overlapping accesses, but deinitialization requires exclusive access; consider copying to a local variable}}
+        // expected-note @-2 {{conflicting access is here}}
+    }
+    f = {}
+    f()
+}
+
+// CHECK-LABEL: sil hidden [ossa] @$s16moveonly_closure026testInOutVarClosureCaptureF0yyyyczF : $@convention(thin) (@inout @callee_guaranteed () -> ()) -> () {
+// CHECK: bb0([[F:%.*]] : $*@callee_guaranteed () -> ()):
+// CHECK: [[BOX:%.*]] = alloc_box ${ var SingleElt }
+// CHECK: [[PROJECT:%.*]] = project_box [[BOX]]
+// CHECK: [[CLOSURE:%.*]] = function_ref @$s16moveonly_closure026testInOutVarClosureCaptureF0yyyyczFyycfU_ : $@convention(thin) (@guaranteed { var SingleElt }) -> ()
+// CHECK: [[BOX_COPY:%.*]] = copy_value [[BOX]]
+// CHECK: [[PAI:%.*]] = partial_apply [callee_guaranteed] [[CLOSURE]]([[BOX_COPY]])
+// CHECK: [[ACCESS:%.*]] = begin_access [modify] [unknown] [[F]]
+// CHECK: assign [[PAI]] to [[ACCESS]]
+// CHECK: end_access [[ACCESS]]
+// CHECK: } // end sil function '$s16moveonly_closure026testInOutVarClosureCaptureF0yyyyczF'
+//
+// CHECK-LABEL: sil private [ossa] @$s16moveonly_closure026testInOutVarClosureCaptureF0yyyyczFyycfU_ : $@convention(thin) (@guaranteed { var SingleElt }) -> () {
+// CHECK: bb0([[BOX:%.*]] :
+// CHECK:   [[PROJECT:%.*]] = project_box [[BOX]]
```

Comment:
Most of the SILGen for the closure bodies appear to have exactly the same pattern (since they have the same bodies), the only difference being whether there's a `project_box` or in one case, no projection because it takes an `(@inout_aliasable SingleElt)`, for the captured var. 

Would it be reasonable to cut down on amount of CHECK's for bits that are already covered by another test, so it's only checking the important parts?

For example, my guess is that this is ensuring there is a `mark_must_check` and then a `load [copy]` and no load_borrow's? If it's possible to cut down on the redundancy, it would make it easier to grok what's important (e.g., even dropping CHECK's for destroy's and end_accesses since they'd be caught by verifier anyway).

Reply:
I am purposely trying to be verbose here so that if someone changes the codegen for any of these individual parts, they need to update this codegen and think about what they are doing. So making the pattern matching looser would go against specifically what I am trying to do.